### PR TITLE
refactor: reimplemented the webview using the new data module and added tests

### PR DESCRIPTION
**Pull request by TomerFi**
- refactor: context web provider re-implemented using the new data module
- test: added test fixtures
- chore: small refactoring for web provider
- test: added test for the context web provider

### CODE REVIEW

**Review by ilan-pinto — APPROVED** [review]

File: src/data/loader.ts [hunk 1]
```diff
@@ -4,13 +4,15 @@ import * as k8s from '@kubernetes/client-node';
 const OCM_GROUP = 'open-cluster-management';
 
 export class OcmResource {
+	readonly kr: any;
```
[the remark this answers is not in the document]
Comment (TomerFi) [reply src/data/loader.ts:7]: kubernetes resource

[the remark this answers is not in the document]
Comment (TomerFi) [reply src/data/loader.ts:7]: I wanted to make sure we encapsulates the sources so we won't limit ourselves with information for the web view.
So every component we have encapsulates it own source for easy access.

I tried to keep the naming convention in line:
- our _ConnectedContext_ encapsulates a kubernetes context under _kontext_.
- our _ConnectedCluster_ encapsulates a kubernetes context under _kluster_.
- our _ConnectedUser_ encapsulates a kubernetes context under _kuser_.
- our _OcmResourceDefinition_ encapsulates a kubernetes crd under _krd_.
- our _OcmResource_ encapsulates a kubernetes cr under _kr_.

[the remark this answers is not in the document]
Comment (TomerFi) [reply src/data/loader.ts:7]: the same goes for our tree view components, eventually, they're the ones triggering the web view, so they encapsulates their own sources as well, which are are own components:
- our _TreeContext_ encapsulates our own _ConnectedCluster_ under _context_.
- our _TreeCrd_ encapsulates our own _OcmResourceDefinition_ under _crd_.
- our _TreeCr_ encapsulates our own _OcmResource_ under _cr_.

That way - no matter what information you require and where are you in the code, you can easily access it.

**Comment by IlonaShishov** [comment]
awesome work! 
/lgtm

**Pull request merged**